ldap/servers/slapd/connection.c

```diff
@@ -271,12 +271,18 @@ connection_reset(Connection *conn, int ns, PRNetAddr *from, int fromLen __attrib
     char buf_destip[INET6_ADDRSTRLEN + 1] = {0};
     char *str_unknown = "unknown";
     int in_referral_mode = config_check_referral_mode();
+    int proxy_connection = 0;
+    PRNetAddr pr_netaddr_dest = {{0}};
 
     slapi_log_err(SLAPI_LOG_CONNS, "connection_reset", "new %sconnection on %d\n", pTmp, conn->c_sd);
 
     /* bump our count of connections and update SNMP stats */
     conn->c_connid = slapi_counter_increment(num_conns);
 
+    if (haproxy_receive(conn->c_sd, &proxy_connection, from, &pr_netaddr_dest) != 0) {
```

Comment:
Just for curiosity, what happens if LDAP client is not HAproxy ? A regular client sending a LDAP bind, can it break its first LDAP request ?

Reply:
About 
> Just for curiosity, what happens if LDAP client is not HAproxy ? A regular client sending a LDAP bind, can it break its first LDAP request ?
haproxy_receive function is quite straightforward:
    the first PDU is read with MSG_PEEK flag (so it stay in the socket queue)
      if it is not a haproxy v1 or an haproxy v2 that is the end of the function.
        (So the answer to @tbordaz question is that the first PDU is still unchanged at that point)
      if it is an haproxy v1 exactly header_len bytes are read from the socket with 0 flag (so the haproxy header is removed from the socket queue)